site/recipes/recipe_different_processing_against_stream.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Recipe 4. Applying Different Processing Against a Single Stream
```

Comment:
Ah, thank you for the clarification. I will update the recipe/code to follow what you expected.